code/modules/power/laser/PTL.dm

```diff
@@ -0,0 +1,981 @@
+/obj/machinery/power/PTL
+	name = "power transmission laser"
+	icon = 'icons/obj/machines/laser.dmi'
+	icon_state = "ptl"
+	desc = "A gigawatt laser that transmits power across vast distances. Don't look into the beam."
+	idle_power_usage = 1000
+	active_power_usage = 1000000
+	var/laser_beam_strength = 1000000
+
+
+
+
+
+
+
+
+/obj/machinery/power/PTL/New()
+	..()
+	var/obj/item/weapon/circuitboard/machine/B = new /obj/item/weapon/circuitboard/machine/PTL(null)
+	B.apply_default_parts(src)
+
+
+
+/obj/machinery/power/PTL/proc/power_beam(dir, strength)
+	var/obj/item/projectile/beam/PTLbeam/P = new /obj/item/projectile/beam/PTLbeam(src.loc)
+	P.power_strength = laser_beam_strength
+	P.speed = 0	//Fast!
+	P.damage = 0	//Calculated in projectile
+	P.nodamage = 0
+	P.legacy = 1
+	P.setDir(src.dir)
+	P.starting = loc
+	P.fire()
+
+
+
+///////////////////////////////
+// General procedures
+//////////////////////////////
+
+// common helper procs for all power machines
+/obj/machinery/power/proc/add_avail(amount)
+	if(powernet)
+		powernet.newavail += amount
+
+/obj/machinery/power/proc/add_load(amount)
+	if(powernet)
+		powernet.load += amount
+
+/obj/machinery/power/proc/surplus()
+	if(powernet)
+		return powernet.avail - powernet.load
+	else
+		return 0
+
+/obj/machinery/power/proc/avail()
+	if(powernet)
+		return powernet.avail
+	else
+		return 0
+
+/obj/machinery/power/proc/disconnect_terminal() // machines without a terminal will just return, no harm no fowl.
+	return
+
+// returns true if the area has power on given channel (or doesn't require power).
+// defaults to power_channel
+/obj/machinery/proc/powered(var/chan = -1) // defaults to power_channel
+	if(!loc)
+		return 0
+	if(!use_power)
+		return 1
+
+	var/area/A = src.loc.loc		// make sure it's in an area
+	if(!A || !isarea(A) || !A.master)
+		return 0					// if not, then not powered
+	if(chan == -1)
+		chan = power_channel
+	return A.master.powered(chan)	// return power status of the area
+
+// increment the power usage stats for an area
+/obj/machinery/proc/use_power(amount, chan = -1) // defaults to power_channel
+	var/area/A = get_area(src)		// make sure it's in an area
+	if(!A || !isarea(A) || !A.master)
+		return
+	if(chan == -1)
+		chan = power_channel
+	A.master.use_power(amount, chan)
+
+/obj/machinery/proc/addStaticPower(value, powerchannel)
+	var/area/A = get_area(src)
+	if(!A || !A.master)
+		return
+	A.master.addStaticPower(value, powerchannel)
+
+/obj/machinery/proc/removeStaticPower(value, powerchannel)
+	addStaticPower(-value, powerchannel)
+
+/obj/machinery/proc/power_change()		// called whenever the power settings of the containing area change
+										// by default, check equipment channel & set flag
+										// can override if needed
+	if(powered(power_channel))
+		stat &= ~NOPOWER
+	else
+
+		stat |= NOPOWER
+	return
+
+// connect the machine to a powernet if a node cable is present on the turf
+/obj/machinery/power/proc/connect_to_network()
```

Comment:
Look, I know this is a wip.

But why the FUCK have you moved all of the power helpers here? that's not happening, fullstop.